.github/workflows/ci-shark-ai.yml

```diff
@@ -54,22 +54,9 @@ jobs:
       - name: Install pip deps
         run: |
           python -m pip install --no-compile --upgrade pip
-          # Note: We install in three steps in order to satisfy requirements
-          # from non default locations first. Installing the PyTorch CPU
-          # wheels saves multiple minutes and a lot of bandwidth on runner setup.
-          pip install --no-compile -r pytorch-cpu-requirements.txt
-          pip install --no-compile -r requirements.txt -e sharktank/ shortfin/
-
-          # Install latest iree-tubrine.
-          pip install --no-compile -f https://iree.dev/pip-release-links.html --src deps \
-            -e "git+https://github.com/iree-org/iree-turbine.git#egg=iree-turbine"
-
-          # Try with the latest IREE nightly releases, not what iree-turbine pins.
-          # We could also pin to a known working or stable version.
-          # This should eventually stabilize. Do the best we can for now.
-          pip install -f https://iree.dev/pip-release-links.html --upgrade --pre \
-            iree-base-compiler \
-            iree-base-runtime
+          pip install shark-ai[apps]
```

Comment:
I would prefer that this test stay as building from source, as opposed to using nightlies for `sharktank/shortfin`, and the pinned version of IREE for those packages.

It's a PR triggered test, and the intent is to catch regressions before they're merged in both sharktank and shortfin. IIUC by using `nightly dev wheels` we would end up catching these failures later (overnight), instead of before they get committed in the first place.

It may be useful though to create a `shark-ai-nightly` ci that essentially repeats the same tests with the nightly dev-wheels,`shark-ai[apps]`, and pinned IREE, because that does add coverage for the code that we are actually releasing. Just feels like it would be a mistake to trade catching potential regressions at the source.

It makes a ton of sense to do this for the `sglang_benchmark_test` though, which is performance regression focused, as opposed to functional regression focused, like this one is.

Reply:
After reading above comments, I do see how it would be helpful to pin to an IREE version and not gate merging PRs

Reply:
We should keep coverage for the latest packages while not blocking development entirely.

Could add a matrix to each job that tests both configurations, then only mark the workflows that use pinned packages "required checks".

Reply:
To confirm, does a nightly/stable `sharktank` installation affect the code it's being tested on? Meaning the code tested is a stale nightly code vs fresh PR code the pre-submit is running on?

Reply:
Ok, I see we want to follow the [developer guide](https://github.com/nod-ai/shark-ai/blob/main/docs/developer_guide.md) instructions for pre-submit and the nightly releases install for nightly CIs. Will revert pre-submit changes.